xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -266,32 +305,47 @@ viewers.Comments = Class.create({
    * Add a preview button that generates the rendered comment,
    */
   addPreview : function(form) {
-    if (!form || !XWiki.hasEdit) {
-      return;
-    }
-    var previewURL = "$xwiki.getURL('__space__.__page__', 'preview')".replace("__space__", encodeURIComponent(XWiki.currentSpace)).replace("__page__", encodeURIComponent(XWiki.currentPage));
-    form.commentElt = form.down('textarea');
-    var buttons = form.down('input[type=submit]').up('div');
-    form.previewButton = new Element('span', {'class' : 'buttonwrapper'}).update(new Element('input', {'type' : 'button', 'class' : 'button', 'value' : "$services.localization.render('core.viewers.comments.preview.button.preview')"}));
-    form.previewButton._x_modePreview = false;
-    form.previewContent = new Element('div', {'class' : 'commentcontent commentPreview'});
-    form.commentElt.insert({'before' : form.previewContent});
-    form.previewContent.hide();
-    buttons.insert({'top' : form.previewButton});
-    form.previewButton.observe('click', function() {
-      if (!form.previewButton._x_modePreview && !form.previewButton.disabled) {
-         form.previewButton.disabled = true;
-         var notification = new XWiki.widgets.Notification("$services.localization.render('core.viewers.comments.preview.inProgress')", "inprogress");
-         new Ajax.Request(previewURL, {
-            method : 'post',
-            parameters : {'xpage' : 'plain', 'sheet' : '', 'content' : form.commentElt.value, 'form_token': form.form_token.value, 'restricted': 'true'},
-            onSuccess : function (response) {
+    require(['jquery'], function ($) {
+      if (!form || !XWiki.hasEdit || jQuery("[name='defaultEditorId']").first().val() !== "text") {
+        return;
+      }
+
+      var previewURL = "$xwiki.getURL('__space__.__page__', 'preview')".replace("__space__",
+          encodeURIComponent(XWiki.currentSpace)).replace("__page__", encodeURIComponent(XWiki.currentPage));
+      form.commentElt = form.down('textarea');
```

Comment:
Even more mixing..

Reply:
I'm not sure why you resolved this conversation. It's still not clear to me why this new code requires jQuery when it continues to use Prototype.js heavily. Why exactly is jQuery required here?

Reply:
I guess I did it by mistake, I just found it back when reviewing again the code... I agree the mix is wrong here. I try to refactor it to jQuery but it's an heavy task. So going to remove the call to jquery instead.